### PR TITLE
Samrooke/ac 1029 remove system args needs from argo task

### PR DESCRIPTION
**Pull request by samrooke**
### Description

Fixes # .

Save default values for argo plugin on the event to database.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] New tests added to cover the changes.
- [x] All tests passed locally.

### CODE REVIEW

**Review by lillie-dae — APPROVED** [review]

File: src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs [hunk 1]
```diff
@@ -70,7 +74,7 @@ public sealed class ArgoPlugin : TaskPluginBase, IAsyncDisposable
             Initialize();
         }
 
-        private void Initialize()
+        private async void Initialize()
```
Comment (lillie-dae) [inline src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs:77]: this should return a task

Comment (lillie-dae) [reply src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs:77]: in fact Initialize is called from the constructer so  youll need to wrap your code in a task.run i think

File: src/TaskManager/Database/TaskDispatchEventRepository.cs [hunk 2]
```diff
@@ -109,5 +109,21 @@ public async Task<bool> RemoveAsync(string taskExecutionId)
                 return false;
             }
         }
+
+        public async Task<TaskDispatchEventInfo> UpdateTaskPluginArgsAsync(TaskDispatchEventInfo taskDispatchEventInfo, Dictionary<string, string> pluginArgs)
+        {
+            Guard.Against.Null(taskDispatchEventInfo, nameof(taskDispatchEventInfo));
```
Comment (lillie-dae) [inline src/TaskManager/Database/TaskDispatchEventRepository.cs:115]: minor thing but dont need to put nameof(taskDispatchEventInfo)) in here

Guard.Against.Null(taskDispatchEventInfo) is good enough and dont risk losing proper name of.

File: src/TaskManager/TaskManager/Services/TaskDispatchEventService.cs [hunk 3]
```diff
@@ -68,5 +68,20 @@ public TaskDispatchEventService(ITaskDispatchEventRepository taskDispatchEventRe
             Guard.Against.NullOrWhiteSpace(taskExecutionId, nameof(taskExecutionId));
             return await _taskDispatchEventRepository.GetByTaskExecutionIdAsync(taskExecutionId).ConfigureAwait(false);
         }
+
+        public async Task<TaskDispatchEventInfo> UpdateTaskPluginArgsAsync(TaskDispatchEventInfo taskDispatchEvent, Dictionary<string, string> pluginArgs)
+        {
+            Guard.Against.Null(taskDispatchEvent, nameof(taskDispatchEvent));
```
Comment (lillie-dae) [inline src/TaskManager/TaskManager/Services/TaskDispatchEventService.cs:74]: another minor thing see above (nameof)

**Review by answeranand — APPROVED** [review]

**Pull request merged**